app/services/activitypub/process_collection_service.rb

```diff
@@ -28,6 +28,14 @@ def different_actor?
     @json['actor'].present? && value_or_id(@json['actor']) != @account.uri
   end
 
+  def suspended_actor?
+    @account.suspended? && !(@account.suspension_origin_remote? && activity_allowed_while_suspended?)
+  end
+
+  def activity_allowed_while_suspended?
+    %w(Delete Update).include?(@json['type']) && (value_or_id(@json['object']) == @account.uri)
```

Comment:
I would allow `Delete` on any object, as well as `Undo Follow` / `Reject Follow` and that kind of stuff, to be honest.

Rationale for `Delete`:
- remote admin suspends their user
- then remove offending posts (while the user is suspended)
- undoes the suspension

Rationale for `Undo Follow` / `Reject Follow`:
- better handling of Unfollow/Refollow flow as suggested in #14775